files.py

```diff
@@ -0,0 +1,120 @@
+"""
+A command line client for interacting with the Cacophony Project Files API.
+
+See https://api.cacophony.org.nz/#api-Files
+"""
+
+# TODO: upload support (with descriptions etc)
+# TODO: sync support (sync file and details between environments)
+
+import argparse
+import sys
+from os.path import join
+
+from api import API
+
+
+def main():
+    args = parse_args()
+    api = API(args.server, args.username, args.password)
+
+    if args.command == "list":
+        do_list(api)
+    elif args.command == "download":
+        do_download(api, args)
+    elif args.command == "delete":
+        do_delete(api, args)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-s", "--server", default="https://api.cacophony.org.nz", help="API server URL"
+    )
+    parser.add_argument("username", help="Username to authenticate with as")
+    parser.add_argument("password", help="Password to authenticate with")
+
+    subparsers = parser.add_subparsers(dest="command")
+
+    # list
+    subparsers.add_parser("list", help="list available files")
+
+    # download
+    download_parser = subparsers.add_parser("download", help="download files")
+    download_parser.add_argument("-i", "--id", help="File id to download")
+    download_parser.add_argument(
+        "--all", action="store_true", help="Download all files"
+    )
+    download_parser.add_argument(
+        "-d", "--directory", default=".", help="Target directory for downloads"
+    )
+
+    # delete
+    delete_parser = subparsers.add_parser("delete", help="delete files")
+    delete_parser.add_argument("id", help="File id to delete", nargs="+")
+
+    args = parser.parse_args()
+    if not args.command:
+        args.command = "list"
+
+    return args
+
+
+def do_list(api):
+    for f in api.list_files():
+        print(format_file(f))
+
+
+def format_file(f):
```

Comment:
perhaps a different name like ```file_to_string``` might be more indicative of the function of this method

Reply:
yep, this name isn't great. I went with `format_file_info`.